src/Internal/Types/ScriptLookups.purs

```diff
@@ -1077,21 +1075,14 @@ processConstraint mpsMap osMap = do
       let value = fromPlutusValue plutusValue
       runExceptT $ _valueSpentBalancesOutputs <>= requireValue value
     MustSpendPubKeyOutput txo -> runExceptT do
-      txOut <- ExceptT $ lookupTxOutRef txo Nothing
-      -- Recall an Ogmios datum is a `Maybe String` where `Nothing` implies a
-      -- wallet address and `Just` as script address.
-      case txOut of
-        TransactionOutput { amount, datum: NoOutputDatum } -> do
-          -- POTENTIAL FIX ME: Plutus has Tx.TxIn and Tx.PubKeyTxIn -- TxIn
-          -- keeps track TransactionInput and TxInType (the input type, whether
-          -- consuming script, public key or simple script)
-          _cpsToTxBody <<< _inputs %= Set.insert txo
-          _valueSpentBalancesInputs <>= provideValue amount
-        _ -> throwError $ TxOutRefWrongType txo
```

Comment:
Why don't we need to throw here anymore?

Reply:
Wallet addresses can contain datum